addons/account/models/account_move.py

```diff
@@ -1361,24 +1361,30 @@ def _compute_partner_credit_warning(self):
                            move.move_type == 'out_invoice' and \
                            move.company_id.account_use_credit_limit
             if show_warning:
-                amount_total_currency = move.currency_id._convert(move.tax_totals['amount_total'], move.company_currency_id, move.company_id, move.date)
-                updated_credit = move.partner_id.commercial_partner_id.credit + amount_total_currency
-                move.partner_credit_warning = self._build_credit_warning_message(move, updated_credit)
+                updated_credit = move.partner_id.commercial_partner_id.credit + move._get_additional_amount()
+                move.partner_credit_warning = self._build_credit_warning_message(
+                    move, updated_credit, move.tax_totals['amount_total'] > 0.0)
 
-    def _build_credit_warning_message(self, record, updated_credit):
-        ''' Build the warning message that will be displayed in a yellow banner on top of the current record
+    def _get_additional_amount(self):
+        # Hook to be overridden to include confirmed Sales Orders.
+        self.ensure_one()
+        return 0.0
+
+    def _build_credit_warning_message(self, record, updated_credit, include):
+        """ Build the warning message that will be displayed in a yellow banner on top of the current record
             if the partner exceeds a credit limit (set on the company or the partner itself).
             :param record:                  The record where the warning will appear (Invoice, Sales Order...).
             :param updated_credit (float):  The partner's updated credit limit including the current record.
+            :param include (bool):          Whether the current record's amount is included in the warning message.
             :return (str):                  The warning message to be showed.
-        '''
+        """
         partner_id = record.partner_id.commercial_partner_id
         if not partner_id.credit_limit or updated_credit <= partner_id.credit_limit:
             return ''
         msg = _('%s has reached its Credit Limit of : %s\nTotal amount due ',
                 partner_id.name,
                 formatLang(self.env, partner_id.credit_limit, currency_obj=record.company_id.currency_id))
-        if updated_credit > partner_id.credit:
+        if include:
```

Comment:
I had to change the condition to make sure the string '(including this document)' was also displayed when creating an invoice from a SO (since the confirmed SOs are now included in the partner's credit) :wink: